test/e2e/datepicker.test.js

```diff
@@ -1,166 +1,168 @@
+    it('als gebruiker kan ik alleen een toegelaten datum kiezen, wanneer ik een foutieve datum kies blijft de datepicker open staan', async () => {
         const datepicker = await vlDatepickerPage.getMinMaxDatepicker();
-        await datepicker.selectMonth('mei');
         await datepicker.selectDay(16);
-
-        await assert.eventually.isTrue(datepicker._isVisible());
+        await assert.eventually.isTrue(datepicker.isOpen());
         await assert.eventually.isEmpty(datepicker.getInputValue());
+        await datepicker.selectDay(15);
+        await assert.eventually.isFalse(datepicker.isOpen());
+        await assert.eventually.equal(datepicker.getInputValue(), '15.05.2019');
     });
 
-    it('ik kan een range selecteren', async () => {
+    it('als gebruiker kan ik een datum range selecteren', async () => {
         const datepicker = await vlDatepickerPage.getRangeDatepicker();
-        
-        await datepicker.selectRange(currentDay, nextDay);
-        
+        await datepicker.selectRange(now.day, tomorrow.day);
         const range = await datepicker.getInputValue();
-        const today = now.toFormat('dd.LL.yyyy');
-        const tomorrow = now.plus({ day: 1 }).toFormat('dd.LL.yyyy');
-        
-        assert.equal(range, today + ' tot ' + tomorrow);
+        assert.equal(range, `${now.toFormat('dd.LL.yyyy')} tot ${tomorrow.toFormat('dd.LL.yyyy')}`);
+    });
+
+    it('als gebruiker kan ik geen datums selecteren die niet beschikbaar zijn', async () => {
+        const datepicker = await vlDatepickerPage.getDisabledDatesDatepicker();
+        await datepicker.selectYear(2019);
+        await datepicker.selectMonth('april');
+        await datepicker.selectDay(1);
+        await assert.eventually.equal(datepicker.getInputValue(), '');
+    });
+
+    it('als gebruiker kan ik een geselecteerde datum lezen in een ander formaat', async () => {
+        const datepicker = await vlDatepickerPage.getAlternatieveVisualisatieDatepicker();
+        await datepicker.selectYear(2018);
+        await datepicker.selectMonth('augustus');
+        await datepicker.selectDay(29);
+        await assert.eventually.equal(datepicker.getVisualisedValue(), 'woensdag 29 aug 2018');
     });
 
-    it('ik kan een tijd selecteren', async () => {
+    it('als gebruiker kan ik een tijd selecteren', async () => {
         const datepicker = await vlDatepickerPage.getTimepicker();
-        await datepicker.selectHour(20);
-        await datepicker.selectMinutes(25);
-        await assert.eventually.equal(datepicker.getInputValue(), '20:25');
+        await datepicker.selectHour('08');
+        await datepicker.selectMinutes('25');
+        await assert.eventually.equal(datepicker.getInputValue(), '08:25');
     });
 
-    it('ik kan een max tijd instellen', async () => {
+    it('als gebruiker kan ik alleen een toegelaten tijd kiezen, wanneer ik een foutieve tijd kies zal de min of max toegelaten tijd gezet worden', async () => {
```

Comment:
ik vermoed dat dit aan webuniversum kant ook zo is, maar dus bij datums kan ik iets wat niet mag niet selecteren, en hier spring em naar het minimum. da lijkt me nie zo consistent.

Reply:
Aan de andere kant zijn het verschillende soorten inputs. Als er bij een klik op een datum niets gebeurd, is dat "normaal". Als je iets intypt en de applicatie gaat ineens uw input leegmaken omdat het fout is (bv. een typo), is dat misschien ambetant (want je wil jezelf kunnen verbeteren).
Aan de andere kant is de gebruiker zijn input zomaar aanpassen ook niet super proper... Ik zou eerder verwachten dat het als invalid wordt aangeduid en dat er een bericht tevoorschijn komt met een melding waarom het invalid is.

Reply:
Nog eens bekeken, en inderdaad consistent met webuniversum. Inderdaad niet helemaal logisch, en zou eventueel gemeld kunnen worden.

Reply:
Ik heb issue #94 aangemaakt om dit (eventueel) later nog eens te bekijken.